### PR TITLE
Drafttest

### PR DESCRIPTION
**Pull request by twsleight**
adding .gitignore, the out_fileread.py script for reading gaussian .out files and the test for the out_fileread.py script.

### CODE REVIEW

File: ames_qsar/out_fileread.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def out_fileread(filepath):
+    import os
+    os.chdir(filepath)
+    #filepath = '/home/twsleight/ames_qsar/tests'
+    import glob
```
Comment (paulboone) [inline ames_qsar/out_fileread.py]: imports should be at top of file

File: ames_qsar/out_fileread.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+def out_fileread(filepath):
+    import os
+    os.chdir(filepath)
+    #filepath = '/home/twsleight/ames_qsar/tests'
+    import glob
+    #print(filepath + '/*.out')
+    namelist = []
+    for filename in glob.glob('*.out'):
+
+    #    print(filename)
```
Comment (paulboone) [inline ames_qsar/out_fileread.py]: debugging prints / comments should not be part of final PR

File: tests/test_fileread.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#adde the current working directory to the PATH
+import sys,os
+sys.path.append(os.getcwd())
```
Comment (paulboone) [inline tests/test_fileread.py]: This should you never need to do.

File: tests/test_fileread.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import sys,os
+sys.path.append(os.getcwd())
+
+import pytest
+import os
+from ames_qsar.out_fileread import out_fileread
+
+def test_fileread():
+    #these are the test files that should be found
+    assert set(out_fileread(os.getcwd()+ '/tests')) == set(['coumarin_gas.out'\
```
Comment (paulboone) [inline tests/test_fileread.py]: there is a correct way of getting package-local files for testing that we will look up

**Review by paulboone — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -4,4 +4,5 @@ build/*
 dist/*
 data/*
 development/*
+amesQSAR.egg-info/
```
Comment (paulboone) [inline .gitignore]: *.egg-info would be more general

File: tests/test_fileread.py [hunk 6]
```diff
@@ -1,12 +1,9 @@
...
-
 import pytest
 import os
 from ames_qsar.out_fileread import out_fileread
 
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
 def test_fileread():
     #these are the test files that should be found
-    assert set(out_fileread(os.getcwd()+ '/tests')) == set(['coumarin_gas.out'\
+    assert set(out_fileread(THIS_DIR)) == set(['coumarin_gas.out'\
```
Comment (paulboone) [inline tests/test_fileread.py:9]: so I just dealt with this for the integration tests I'm working on. You will want to:
* use the `tmpdir` fixture
* cd to the tmpdir at the beginning of the test (so the files get created there)
* check for their existence there.
I will go over all this on Monday, too.

File: tests/test_fileread.py [hunk 7]
```diff
@@ -1,12 +1,9 @@
-#adde the current working directory to the PATH
-import sys,os
-sys.path.append(os.getcwd())
-
 import pytest
 import os
 from ames_qsar.out_fileread import out_fileread
```
Comment (paulboone) [inline tests/test_fileread.py:5]: pep8 module ordering:

Imports should be grouped in the following order:

    Standard library imports.
    Related third party imports.
    Local application/library specific imports.

You should put a blank line between each group of imports.

File: ames_qsar/out_fileread.py [hunk 8]
```diff
@@ -1,17 +1,11 @@
 import os
 import glob
 def out_fileread(filepath):
```
Comment (paulboone) [inline ames_qsar/out_fileread.py:4]: add blank line between imports and def

**Review by paulboone — CHANGES_REQUESTED** [review]
So new commits are clearly named and it easy to follow. Since this is all new code though, you can squash some of the changes:

* "adding .egg-info to .gitignore" can go in initial .gitignore commit
* "removing whitespace and debugging code" can be squashed in to where you added the file

If we had already merged your prior commits, then the new atomic commits would be perfect for making the fixes. Since we haven't merged, we have the opportunity to clean up the commits so it looks like you did it right the first time. :)

See other comments.

File: tests/test_fileread.py (hunk 6 above)
Comment (twsleight) [reply tests/test_fileread.py:9]: In my case these files aren't actually created by the script, they're already in existence in the test folder. The test checks the functions ability to find them all.

**Pull request closed without merging**